File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def dec2n(n):
    cnt = 0
    s = ''
    while n > 0:
        r = n % 2
        if r == 1: cnt += 1
        s = str(r) + s
        n //= 2
    return cnt

def solution(n):
    cnt = dec2n(n)
    for num in range(n+1,10000000):
        if dec2n(num) == cnt:
            return num

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int cnt = dec2n(n);
        int ans = 0;
        for (int i = n+1; i < 1000000000; i++) {
            if (cnt == dec2n(i)) {
                ans = i;
                break;
            }
        }
        return ans;
    }
    public int dec2n(int n) {
        int cnt = 0;
        while (n > 0) {
            int r = n % 2;
            if (r == 1) cnt++;
            n /= 2;
        }
        return cnt;
    }
}